frontend/src/component/changeRequest/ChangeRequest/Changes/Change/StrategyChangeOverwriteWarning.tsx

```diff
@@ -73,10 +75,18 @@ export const ChangesToOverwrite: React.FC<{
     change: IChangeRequestUpdateStrategy;
 }> = ({ change, currentStrategy }) => {
     const checkForChanges = useUiFlag('changeRequestConflictHandling');
+    const { registerWillOverwriteStrategyChanges } =
+        useChangeRequestPlausibleContext();
     const changesThatWouldBeOverwritten = checkForChanges
         ? getChangesThatWouldBeOverwritten(currentStrategy, change)
         : null;
 
+    useEffect(() => {
+        if (changesThatWouldBeOverwritten) {
+            registerWillOverwriteStrategyChanges();
+        }
+    }, [changesThatWouldBeOverwritten]);
```

Comment:
This needs to be wrapped in `useEffect` to avoid the issue where react complains you cannot updated a component while rendering another component. Refer to [this blog post](https://bobbyhadz.com/blog/react-cannot-update-component-while-rendering-different-component).

There may also be other ways of solving this, but this was the first one I found.